aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_bedrock_patches.py

```diff
@@ -0,0 +1,325 @@
+        "aws.bedrock.datasource_id": "dataSourceId",
+    }
+
+    @classmethod
+    def operation_names(cls):
+        return ["DeleteDataSource", "GetDataSource", "UpdateDataSource"]
+
+
+_OPERATION_MAPPING = {
+    op_name: op_class
+    for op_class in [_KnowledgeBaseOperation, _DataSourceOperation, _AgentOperation]
+    for op_name in op_class.operation_names()
+    if inspect.isclass(op_class) and issubclass(op_class, _BedrockAgentOperation) and not inspect.isabstract(op_class)
+}
+
+
+class _BedrockAgentExtension(_AwsSdkExtension):
+    def __init__(self, call_context: _AwsSdkCallContext):
+        super().__init__(call_context)
+        self._op = _OPERATION_MAPPING.get(call_context.operation)
+
+    def extract_attributes(self, attributes: _AttributeMapT):
+        if self._op is None:
+            return
+        for key, value in self._op.start_attributes.items():
+            extracted_value = self._call_context.params.get(value)
+            if extracted_value:
+                attributes[key] = extracted_value
+
+    def on_success(self, span: Span, result: _BotoResultT):
+        if self._op is None:
+            return
+
+        for key, value in self._op.response_attributes.items():
+            response_value = result.get(value)
+            if response_value:
+                span.set_attribute(
+                    key,
+                    response_value,
+                )
+
+
+class _BedrockAgentRuntimeExtension(_AwsSdkExtension):
+    def extract_attributes(self, attributes: _AttributeMapT):
+        agent_id = self._call_context.params.get("agentId")
+        if agent_id:
+            attributes["aws.bedrock.agent_id"] = agent_id
+
+        knowledgebase_id = self._call_context.params.get("knowledgeBaseId")
+        if knowledgebase_id:
+            attributes["aws.bedrock.knowledgebase_id"] = knowledgebase_id
+
+
+class _BedrockExtension(_AwsSdkExtension):
+    # pylint: disable=no-self-use
+    def on_success(self, span: Span, result: _BotoResultT):
+        # GuardrailId
+        guardrail_id = result.get("guardrailId")
+        if guardrail_id:
+            span.set_attribute(
+                "aws.bedrock.guardrail_id",
+                guardrail_id,
+            )
+
+
+class BaseBedrockRuntimeModel(abc.ABC):
+    @classmethod
+    def model_name(cls):
+        pass
+
+    @classmethod
+    def extract_attributes(cls, context_param: Dict[str, Any], attributes: _AttributeMapT):
+        pass
+
+    @classmethod
+    def on_success(cls, span: Span, result: _BotoResultT):
+        pass
+
+
+class TitanBedrockRuntimeModel(BaseBedrockRuntimeModel):
+    @classmethod
+    def model_name(cls):
+        return "amazon.titan"
+
+    @classmethod
+    def extract_attributes(cls, context_param: Dict[str, Any], attributes: _AttributeMapT):
+        if "body" not in context_param:
+            return
+
+        text_generation_config = json.loads(context_param.get("body")).get("textGenerationConfig")
```

Comment:
textGenerationConfig is the define promot format for titan model:https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-titan-text.html
gen_ai.prompt have security concern since if might contains customer info. 
gen_ai.response.finish_reasons: We have some concern on extraction method, so we on hold of this attributes, we have cut a ticket to updatream: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2601